queries/reflect.go

```diff
@@ -301,24 +302,32 @@ func PtrsFromMapping(val reflect.Value, mapping []uint64) []interface{} {
 func ValuesFromMapping(val reflect.Value, mapping []uint64) []interface{} {
 	ptrs := make([]interface{}, len(mapping))
 	for i, m := range mapping {
-		ptrs[i] = ptrFromMapping(val, m, false).Interface()
+		ptrs[i] = ptrFromMapping(val, m, false)
 	}
 	return ptrs
 }
 
 // ptrFromMapping expects to be passed an addressable struct that it's looking
 // for things on.
-func ptrFromMapping(val reflect.Value, mapping uint64, addressOf bool) reflect.Value {
+func ptrFromMapping(val reflect.Value, mapping uint64, addressOf bool) interface{} {
+
+	if mapping == uint64(0) {
```

Comment:
You should not need `uint64(0)`, it should be fine if you remove the conversion since Go constants are magical.

Reply:
I will change to "if mapping == 0" as your suggestion.